app/screens/auth/login/LoginScreen.js

```diff
@@ -0,0 +1,191 @@
+import {
+    Image,
+    View,
+    Text,
+    TextInput,
+    TouchableOpacity,
+    KeyboardAvoidingView,
+    Keyboard
+} from 'react-native';
+
+import appLogo from '../../../assets/logo.jpg';
+import AuthStyles from '../AuthStyles';
+import AuthService from '../../../services/auth/AuthService';
+
+class LoginScreen extends Component {
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            username: '',
+            password: '',
+            error: '',
+            usernameTouched: false,
+            passwordTouched: false,
+            formStyle: AuthStyles.form,
+            validationError: ''
+        }
+
+    }
+
+    handleUsernameChange = username => {
+        username = username.split(' ').join('');
+
+        this.setState({
+            username: username
+        });
+    }
+
+    handleUsernameBlur = () => {
+        this.setState({
+            usernameTouched: true
+        });
+    }
+
+    handlePasswordChange = password => {
+        this.setState({
+            password: password
+        });
+    }
+
+    handlePasswordBlur = () => {
+        this.setState({
+            passwordTouched: true
+        });
+    }
+
+    handleFormStyle = () => {
+        this.setState({
+            formStyle: {
+                ...AuthStyles.form,
+                marginTop: 100
+            }
+        });
+    }
+
+    handleLoginPress = async () => {
+
+        await AuthService.login({ username: this.state.username.toLowerCase(), password: this.state.password })
+            .then(response => {
+                if (response === false) {
+                    Keyboard.dismiss();
+                    this.setState({
+                        validationError: 'The credentials do not match with our records.'
+                    });
+                } else {
+                    this.props.navigation.navigate('PokemonList');
+                }
+
+            });
+    }
+
+    componentWillMount() {
+        this.keyboardDidShowListener = Keyboard.addListener('keyboardDidShow', this._keyboardDidShow);
+        this.keyboardDidHideListener = Keyboard.addListener('keyboardDidHide', this._keyboardDidHide.bind(this));
+    }
+
+    componentWillUnmount() {
+        this.keyboardDidShowListener.remove();
+        this.keyboardDidHideListener.remove();
+
+        this.setState({
+            username: '',
+            password: '',
+            error: '',
+            usernameTouched: false,
+            passwordTouched: false,
+            formStyle: AuthStyles.form,
+            validationError: ''
+        });
+    }
+
+    _keyboardDidHide() {
+        this.setState({
+            formStyle: {
+                ...AuthStyles.form,
+                marginTop: 0
+            }
+        });
+    }
+
+    render() {
+
+        const { username, password, usernameTouched, passwordTouched } = this.state;
+
+        const usernameError = !username && usernameTouched ? 'The username is required' : undefined;

```

Comment:
you can do `!username && usernameTouched && 'The username is required'` and will have the same result